privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/ProtectionStatusViewStateMapper.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.privacy.dashboard.impl.ui
+
+import com.duckduckgo.app.global.model.Site
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.privacy.config.api.ContentBlocking
+import com.duckduckgo.privacy.config.api.PrivacyFeatureName.ContentBlockingFeatureName
+import com.duckduckgo.privacy.config.api.UnprotectedTemporary
+import com.duckduckgo.privacy.dashboard.impl.ui.PrivacyDashboardHybridViewModel.ProtectionStatusViewState
+import com.duckduckgo.privacy.dashboard.impl.ui.PrivacyDashboardHybridViewModel.State.*
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface ProtectionStatusViewStateMapper {
+    fun mapFromSite(site: Site): ProtectionStatusViewState
+}
+
+@ContributesBinding(AppScope::class)
+class AppProtectionStatusViewStateMapper @Inject constructor(
+    private val contentBlocking: ContentBlocking,
+    private val unprotectedTemporary: UnprotectedTemporary
+) : ProtectionStatusViewStateMapper {
+
+    override fun mapFromSite(site: Site): ProtectionStatusViewState {
+        val enabledFeatures = mutableListOf<String>().apply {
+            if (!contentBlocking.isAnException(site.url)) {
+                add(ContentBlockingFeatureName.value)
```

Comment:
I don't fully understand why this is needed? Perhaps a comment explaining would be useful.

The other part is that I don't think this scales. If you needed to add 40 features the constructor wouldn't be very nice :D. Probably a plugin to provide these values would be better. I suggest for now just leaving a comment with the plugin idea in case we come back here in the future to add more features.

Reply:
gotcha, I know what you mean. Yes, if at some point the number of privacy features we need to depend on here grows, we could change the way we provide this list. Using plugins is an approach I also like, so I will leave a comment. Right now, it's only one feature, so it would be premature to extract that part.

I will leave a comment linking the docs with a brief explanation. "List of enabled features that are supported by the privacy dashboard".  (right now only contentblocking) https://duckduckgo.github.io/privacy-dashboard/example/docs/interfaces/Generated_Schema_Definitions.ProtectionsStatus.html#enabledFeatures